src/oldreader_api.cpp

```diff
@@ -66,24 +66,26 @@ std::string oldreader_api::retrieve_auth() {
 	}
 
 	std::string pass = cfg->get_configvalue("oldreader-password");
-    std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
-    std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
+	std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
+	std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
 	if ( pass == "" ) {
-        if ( pass_file != "" ) {
-            pass = read_password(pass_file.c_str());
-        } else if ( pass_eval != "") {
-            pass = eval_password(pass_eval.c_str());
-        }
+		if ( pass_file != "" ) {
+			pass = read_password(pass_file);
+			if ( pass == "" ) return "";
+		} else if ( pass_eval != "") {
+			pass = eval_password(pass_eval);
+			if ( pass == "" ) return "";
```

Comment:
No problem, I had the same issue with one of your replies and only saw them after checking my mail. Thought you were busy and was at work myself ;)

Since you misunderstood the code and I had to reread it a lot of times, maybe this is what the code does but not what it's intended to do. Maybe it would make more sense to have it read `password` and if this is empty check either `passwordfile`, `passwordeval` or read from cin if neither is set and if this didn't work return (since it doesn't make sense to authenticate with an empty string as password). I don't know, but anyway the rewrite should at least make the code a bit more straight forward.